marge/job.py

```diff
@@ -151,6 +151,37 @@ def rebase_and_accept(self, approvals):
             except gitlab.Unauthorized:
                 log.warning('Unauthorized!')
                 raise CannotMerge('My user cannot accept merge requests!')
+            except gitlab.NotFound as e:
+                log.warning('Not Found!: %s', e)
+                merge_request.refetch_info()
+                if merge_request.state == 'merged':
+                    # someone must have hit "merge when build succeeds" and we lost the race,
+                    # the branch is gone and we got a 404. Anyway, our job here is done.
```

Comment:
Maybe log it when it happens? Marge got blamed for missing Reviewer-tags before because she was assigned but someone pressed merge :)

(Maybe worth moving the log line above down into the `then` arm and be more specific and adapt else branch, too)